cmd/configure_test.go

```diff
@@ -1,106 +1,82 @@
 			existingAPICfg: nil,
 			expectedAPICfg: &config.APIConfig{BaseURL: "http://example.com"},
 		},
 		{
-			// It overwrites the flags in the config file.
+			desc:           "It overwrites the flags in the config file.",
 			args:           []string{"fakeapp", "configure", "--token", "b", "--workspace", "/b", "--api", "http://example.com/v2"},
 			existingUsrCfg: &config.UserConfig{Token: "token-b", Workspace: "/workspace-b"},
 			expectedUsrCfg: &config.UserConfig{Token: "b", Workspace: "/b"},
 			existingAPICfg: &config.APIConfig{BaseURL: "http://example.com/v1"},
 			expectedAPICfg: &config.APIConfig{BaseURL: "http://example.com/v2"},
 		},
 		{
-			args: []string{"fakeapp", "configure", "--token", "c"},
-			// It overwrites the flags that are passed without losing the ones that are not.
+			desc:           "It overwrites the flags that are passed, without losing the ones that are not.",
+			args:           []string{"fakeapp", "configure", "--token", "c"},
 			existingUsrCfg: &config.UserConfig{Token: "token-c", Workspace: "/workspace-c"},
 			expectedUsrCfg: &config.UserConfig{Token: "c", Workspace: "/workspace-c"},
-			// It gets the default API base URL.
+		},
+		{
+			desc:           "It gets the default API base URL.",
+			args:           []string{"fakeapp", "configure"},
 			existingAPICfg: &config.APIConfig{},
 			expectedAPICfg: &config.APIConfig{BaseURL: "https://api.exercism.com/v1"},
 		},
 	}
 
-	for i, test := range tests {
-		// Create a fake config dir.
-		dir, err := ioutil.TempDir("", fmt.Sprintf("user-config-%d", i))
-		assert.NoError(t, err)
-		defer os.RemoveAll(dir)
-
-		// Override the environment to use the fake config dir.
-		os.Setenv(cfgHomeKey, dir)
+	for _, test := range tests {
+		cmdTest := &CommandTest{
+			Cmd:    configureCmd,
+			InitFn: initConfigureCmd,
+			Args:   test.args,
+		}
+		cmdTest.Setup(t)
+		defer cmdTest.Teardown(t)
 
 		if test.existingUsrCfg != nil {
 			// Write a fake config.
 			cfg := config.NewEmptyUserConfig()
 			cfg.Token = test.existingUsrCfg.Token
 			cfg.Workspace = test.existingUsrCfg.Workspace
-			err = cfg.Write()
-			assert.NoError(t, err)
+			err := cfg.Write()
+			assert.NoError(t, err, test.desc)
 		}
 
-		// Fake out the command-line arguments with the correct subcommand.
-		os.Args = test.args
+		cmdTest.App.Execute()
 
-		// Re-initialize the command so it picks up the fake environment.
-		configureCmd.ResetFlags()
-		// Rerun the config initialization so that the flags get bound properly.
-		initConfigureCmd()
-
-		// Finally. Execute the configure command.
-		fakeCmd.Execute()
-
-		// Now let's get a new config and see that it got written properly.
-		usrCfg, err := config.NewUserConfig()
-		assert.NoError(t, err)
-
-		assert.Equal(t, test.expectedUsrCfg.Token, usrCfg.Token)
-		assert.Equal(t, test.expectedUsrCfg.Workspace, usrCfg.Workspace)
-
-		apiCfg, err := config.NewAPIConfig()
-		assert.NoError(t, err)
+		if test.expectedUsrCfg != nil {
```

Comment:
I had to wrap the sets of assertions in an if statement so that I could split apart the unrelated test cases.